### PR TITLE
Improve messaging when dependencies conflict

### LINKED ISSUE #5260: Running `pod install` with an outdated master specs repo shows error message twice
* [x] I've co-authored and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

> ℹ Please replace this with what you did.  

```sh
   eigen   master ✔  bundle exec pod install                                                                                                                                                                                                                                                                 11:17:44
Analyzing dependencies
Pre-downloading: `Artsy+UIFonts` from `https://github.com/artsy/Artsy-UIFonts.git`, commit `6295cce382764bbca9f2b3e972f71eb2b4d32d57`
Pre-downloading: `FBSnapshotTestCase` from `https://github.com/orta/ios-snapshot-test-case.git`, commit `6b3e8978084388f9aae392fd1e0e21b1ddb88ae3`
Pre-downloading: `FLKAutoLayout` from `https://github.com/orta/FLKAutoLayout.git`, commit `4aa27fa1a9aba36d5465e6cd181ba4b78b0691d3`
Pre-downloading: `Interstellar` from `https://github.com/ashfurrow/Interstellar.git`, commit `3785de4184bd2b61761684d0be07804809681bb3`
Fetching podspec for `Keys` from `Pods/CocoaPodsKeys`
[!] Unable to satisfy the following requirements:

- `Nimble-Snapshots` required by `Podfile`
- `Nimble-Snapshots (= 4.0.1)` required by `Podfile.lock`

None of your spec sources contain a spec satisfying the dependency: `Nimble-Snapshots`.

You have either:
 * out-of-date source repos which you can update with `pod repo update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.

None of your spec sources contain a spec satisfying the dependency: `Nimble-Snapshots (= 4.0.1)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.
```

## What did you expected to happen?

```sh
   eigen   master ✔  bundle exec pod install                                                                                                                                                                                                                                                                 11:17:44
Analyzing dependencies
Pre-downloading: `Artsy+UIFonts` from `https://github.com/artsy/Artsy-UIFonts.git`, commit `6295cce382764bbca9f2b3e972f71eb2b4d32d57`
Pre-downloading: `FBSnapshotTestCase` from `https://github.com/orta/ios-snapshot-test-case.git`, commit `6b3e8978084388f9aae392fd1e0e21b1ddb88ae3`
Pre-downloading: `FLKAutoLayout` from `https://github.com/orta/FLKAutoLayout.git`, commit `4aa27fa1a9aba36d5465e6cd181ba4b78b0691d3`
Pre-downloading: `Interstellar` from `https://github.com/ashfurrow/Interstellar.git`, commit `3785de4184bd2b61761684d0be07804809681bb3`
Fetching podspec for `Keys` from `Pods/CocoaPodsKeys`
[!] Unable to satisfy the following requirements:

- `Nimble-Snapshots` required by `Podfile`
- `Nimble-Snapshots (= 4.0.1)` required by `Podfile.lock`

None of your spec sources contain a spec satisfying the dependency: `Nimble-Snapshots`.

You have either:
 * out-of-date source repos which you can update with `pod repo update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.
```

Perhaps we need some logic checking if there is a dependency with with no "version", and the same one with a version?

### PR DESCRIPTION
**Pull request by orta**
Fixes #5260 

Does this feel right? Can do tests etc then

### CODE REVIEW

File: lib/cocoapods/resolver.rb [hunk 1]
```diff
@@ -419,20 +419,21 @@ def handle_resolver_error(error)
...
-              else
-                message << "\n\nSpecs satisfying the `#{r}` dependency were found, " \
-                  'but they required a higher minimum deployment target.'
-              end
+            conflicts = conflict.requirements.values.flatten.uniq
+            all_cannot_be_found = conflicts.find { |conflict| search_for(conflict).empty? }
+            if all_cannot_be_found
+              # There are no existing specification inside any of the spec repos with given requirements.
+              dependencies = conflicts.count == 1 ? "dependency" : "dependencies"
+              message << "\n\nNone of your spec sources contain a spec satisfying the #{dependencies}: `#{conflicts.join(" ,")}`." \
```
Comment (segiddins) [inline lib/cocoapods/resolver.rb]: `conflicts.join(', ')`

**Comment by segiddins** [comment]
👍 with tests

**Comment by orta** [comment]
Alright - that should do it 👍

**Comment by segiddins** [comment]
👍 good to merge once travis runs

**Comment by orta** [comment]
This is failing at bundler level: `An error occurred while installing json (1.7.7), and Bundler cannot continue.
`

**Comment by segiddins** [comment]
I think this is due to the update to 7.3.1, and I won't have time to investigate this 😔

**Pull request merged**